scipy/_lib/_util.py

```diff
@@ -815,15 +815,16 @@ def _rng_spawn(rng, n_children):
     return child_rngs
 
 
-def _get_nan(*data):
+def _get_nan(*data, xp=None):
+    xp = array_namespace(*data) if xp is None else xp
     # Get NaN of appropriate dtype for data
-    data = [np.asarray(item) for item in data]
+    data = [xp.asarray(item) for item in data]
     try:
-        dtype = np.result_type(*data, np.half)  # must be a float16 at least
+        dtype = xp.result_type(*data, xp.float32)  # must be a float16 at least
```

Comment:
array_api_strict doesn't have a `float16`. Making this change would mean that stats functions with NaNs and `float16` input would return `float32` output. Ideas for special-casing this?  Create an `is_array_api_strict` function?

Reply:
That sounds wrong. We need the general case to be array-agnostic, hence we can't rely on float16 support. Can we use `is_numpy(xp)` and support float16 in the special case?

Reply:
Oh, so going forward, we only support `float16` for NumPy, not if the array-API library happens to support it? I'm OK with that.

Reply:
Seems alright to me. I would say the priority is just getting things done that work in general, but when these things are easy one-liners, sure, seems worth it.